src/Nethermind/Nethermind.Merge.Plugin/Handlers/ExchangeCapabilitiesHandler.cs

```diff
@@ -13,7 +14,7 @@ namespace Nethermind.Merge.Plugin.Handlers;
 
 public class ExchangeCapabilitiesHandler : IAsyncHandler<IEnumerable<string>, IEnumerable<string>>
 {
-    private static IDictionary<string, bool> _capabilities = new Dictionary<string, bool>();
+    private static IDictionary<string, bool> _capabilities = new ConcurrentDictionary<string, bool>();
```

Comment:
Why concurrent?

Reply:
There were race condition issues in our tests. I presume this could happen in the production as well.

Reply:
But isn't it readonly outside the constructor?